### PR TITLE
Update challenge

### PR DESCRIPTION
**Pull request by arechsteiner**
Update to Next.js 14 from 12, add prettier, update rest of the packages.

### CODE REVIEW

File: hooks/use-api-data.ts [hunk 1]
```diff
@@ -1,13 +1,16 @@
...
-    axios.get<T>(path).catch(err => err.response).then(response => {
-      setData(response.data)
-    })
+    axios
+      .get<T>(path)
+      .catch((err) => err.response)
+      .then((response) => {
+        setData(response.data)
+      })
   }, [])
```
Comment (jonatanramhoj) [inline hooks/use-api-data.ts:14]: I get `missing dependency: 'path'`

File: pages/index.tsx [hunk 2]
```diff
@@ -1,30 +1,32 @@
...
-          <div className='ml-auto text-mono'>
-            {airport.country}
-          </div>
-        </a>
-      ))}
-    </div>
-  </Layout>
+      <div>
+        {airports.map((airport) => (
+          <a
```
Comment (jonatanramhoj) [inline pages/index.tsx:16]: Can use `Link` here from `next/link`

**Review by jonatanramhoj — APPROVED** [review]
LGTM with two small comments

File: pages/index.tsx (hunk 2 above)
Comment (arechsteiner) [reply pages/index.tsx]: Nice catch, I was thinking it felt a bit off.

File: hooks/use-api-data.ts (hunk 1 above)
Comment (arechsteiner) [reply hooks/use-api-data.ts:14]: That's a little weird because path is a parameter of the function. `yarn tsc` / `yarn lint` is clear.

**Pull request merged**